gno.land/pkg/sdk/vm/convert.go

```diff
@@ -153,6 +139,22 @@ func convertArgToGno(arg string, argT gno.Type) (tv gno.TypedValue) {
 			}
 			tv.SetUint64(u64)
 			return
+		case gno.Float32Type:
+			assertCharNotPlus(arg[0])
+			f32, err := strconv.ParseFloat(arg, 32)
```

Comment:
I'm not sure this is the correct way of doing this. Essentially, it differs from the way the GnoVM parses floats. Ideally, I think we should have float parsing here be done in the same way as it is done in the GnoVM (parse the basic literal as a constant expression, then convert to a type). This is the only way to make sure there are no inconsistencies when doing `MyFunction(4.5)` from Gno code and doing maketx call with `4.5`.

Reply:
I've made it use `apd` but I'm not sure if it will have an effect since it also uses `strconv` to parse its own string representation to produce float values. Though there may be some difference in how it parses the string value initially; this I'm not sure about.